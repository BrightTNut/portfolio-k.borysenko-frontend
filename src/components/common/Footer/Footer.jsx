import s from './Footer.module.css';

const Footer = () => {
  return (
    <footer>
      <div className={s.container}>Footer</div>
    </footer>
  );
};

export default Footer;
