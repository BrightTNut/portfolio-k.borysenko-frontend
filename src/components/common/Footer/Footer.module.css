.container {
  padding: 15px 30px;
  box-sizing: border-box;
  background-color: var(--bg-primary);
  background-color: var(--bg-tertiary);
  color: var(--text-inverse);
  height: 150px;
}
