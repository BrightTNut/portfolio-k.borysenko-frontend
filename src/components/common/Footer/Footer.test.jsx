import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Footer from './Footer';
import {
  EMAIL,
  LOCATION,
  GITHUB_URL,
  MIAMI_MAP_URL,
} from '../../../utils/constants';

describe('Footer Component', () => {
  it('contains the correct email link', () => {
    render(<Footer />);
    const emailLink = screen.getByRole('link', { name: EMAIL });
    expect(emailLink).toHaveAttribute('href', `mailto:${EMAIL}`);
  });

  it('contains the correct location link', () => {
    render(<Footer />);
    const locationLink = screen.getByRole('link', { name: LOCATION });
    expect(locationLink).toHaveAttribute('href', MIAMI_MAP_URL);
    expect(locationLink).toHaveAttribute('target', '_blank');
    expect(locationLink).toHaveAttribute('rel', 'noopener noreferrer');
  });

  it('contains the correct GitHub link', () => {
    render(<Footer />);
    const githubLink = screen.getByRole('link', { name: /ideas in code/i });
    expect(githubLink).toHaveAttribute('href', GITHUB_URL);
    expect(githubLink).toHaveAttribute('target', '_blank');
    expect(githubLink).toHaveAttribute('rel', 'noopener noreferrer');
  });
});
