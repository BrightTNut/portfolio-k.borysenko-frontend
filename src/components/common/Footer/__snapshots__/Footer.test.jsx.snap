// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches snapshot 1`] = `
<DocumentFragment>
  <footer>
    <div
      style="height: 3px; background-color: rgb(255, 255, 255);"
    />
    <div
      class="container"
    >
      <a
        class="logo_link"
        href="https://github.com/Kateryna-Borysenko"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Logo"
          src="logo.png"
          width="100px"
        />
        <p
          class="logo_description"
        >
          IDEAS IN CODE
        </p>
      </a>
      <div
        class="copyright_info"
      >
        Â© 2023 Portfolio. All rights reserved.
      </div>
      <div>
        <div
          class="title"
        >
          CONTACTS
        </div>
        <a
          class="cell_phone"
          href="tel:18888888888"
        >
          +1 (888) 888-8888
        </a>
        <div
          class="email"
        >
          <a
            href="mailto:borysenko.info@gmail.com"
          >
            borysenko.info@gmail.com
          </a>
        </div>
        <div
          class="container"
        >
          Local time: 8:05 AM
        </div>
        <div
          class="location"
        >
          <a
            href="https://www.google.com/maps/place/Miami,+FL"
            rel="noopener noreferrer"
            target="_blank"
          >
            Miami, Florida
          </a>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
